src/integration_test/java/com/klodnicki/Bike/v2/rest/controller/AdminChargingStationControllerIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+                        .queryParam("bikeId", bike.getId())
+                        .build())
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(ChargingStation.class)
+                .consumeWith(response -> {
+                            ChargingStation station = response.getResponseBody();
+                            assertNotNull(station);
+                            assertFalse(station.getBikeList().isEmpty());
+                            assertIterableEquals(chargingStation.getBikeList(), station.getBikeList());
+                        }
+                );
+    }
+
+    @Test
+    public void add_ShouldAddChargingStationToDatabaseAndReturnStationForAdminResponseDTO_WhenChargingStationIsGiven() {
+        StationForAdminResponseDTO expected = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+
+        webTestClient.post()
+                .uri("/api/admin/stations/add")
+                .bodyValue(chargingStation)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(StationForAdminResponseDTO.class)
+                .consumeWith(response -> {
+                            StationForAdminResponseDTO stationDTO = response.getResponseBody();
+                            assertNotNull(stationDTO);
+                            assertEquals(expected.getId(), stationDTO.getId());
+                            assertEquals(expected.getName(), stationDTO.getName());
+                            assertEquals(expected.getAddress(), stationDTO.getAddress());
+                            assertEquals(expected.getCity(), stationDTO.getCity());
+                            assertEquals(expected.getFreeSlots(), stationDTO.getFreeSlots());
+                        }
+                );
+    }
+
+    @Test
+    public void findById_ShouldReturnStationForAdminResponseDTO_WhenStationIdIsGivenAndStationExistsInDatabase() {
+
+        webTestClient.get()
+                .uri("/api/admin/stations/" + chargingStation.getId())
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(StationForAdminResponseDTO.class)
+                .consumeWith(response -> {
+                    StationForAdminResponseDTO stationDTO = response.getResponseBody();
+                    assertNotNull(stationDTO);
+                    assertEquals(chargingStation.getId(), stationDTO.getId());
+                    assertEquals(chargingStation.getName(), stationDTO.getName());
+                    assertEquals(chargingStation.getAddress(), stationDTO.getAddress());
+                    assertEquals(chargingStation.getCity(), stationDTO.getCity());
+                    assertEquals(chargingStation.getFreeSlots(), stationDTO.getFreeSlots());
+                });
+    }
+
+    @Test
+    public void findAll_ShouldReturnOneObjectCalledListStationsForAdminResponseDTO_WhenStationsExistInDatabase() {
```

Comment:
Here the naming is also invalid. What does this should mean? I bet you can rework it to better reflect the tests.

Reply:
what do you think about this?
findAll_MapsAndReturnsListOfStationsForAdminResponseDTO_WhenStationsExistInDatabase()